libraries/AP_Motors/AP_MotorsHeli_Dual.cpp

```diff
@@ -215,59 +215,43 @@ void AP_MotorsHeli_Dual::set_update_rate( uint16_t speed_hz )
     _speed_hz = speed_hz;
 
     // setup fast channels
-    uint32_t mask = 0;
-    for (uint8_t i=0; i<AP_MOTORS_HELI_DUAL_NUM_SWASHPLATE_SERVOS; i++) {
-        mask |= 1U << (AP_MOTORS_MOT_1+i);
-    }
-    if (_swashplate1.get_swash_type() == SWASHPLATE_TYPE_H4_90 || _swashplate1.get_swash_type() == SWASHPLATE_TYPE_H4_45) {
-        mask |= 1U << (AP_MOTORS_MOT_7);
-    }
-    if (_swashplate2.get_swash_type() == SWASHPLATE_TYPE_H4_90 || _swashplate2.get_swash_type() == SWASHPLATE_TYPE_H4_45) {
-        mask |= 1U << (AP_MOTORS_MOT_8);
-    }
-
+    uint32_t mask = get_swashplate_servo_mask();
     rc_set_freq(mask, _speed_hz);
 }
 
 // init_outputs
 bool AP_MotorsHeli_Dual::init_outputs()
 {
+    // run swashplate configuration
     if (!initialised_ok()) {
-        // make sure 6 output channels are mapped
-        for (uint8_t i=0; i<AP_MOTORS_HELI_DUAL_NUM_SWASHPLATE_SERVOS; i++) {
-            add_motor_num(CH_1+i);
-        }
-        if (_swashplate1.get_swash_type() == SWASHPLATE_TYPE_H4_90 || _swashplate1.get_swash_type() == SWASHPLATE_TYPE_H4_45) {
-            add_motor_num(CH_7);
+        for (uint8_t s=0; s<AP_MOTORS_HELI_N_SWASH; s++) {
+            _swashplate[s].configure();
         }
-        if (_swashplate2.get_swash_type() == SWASHPLATE_TYPE_H4_90 || _swashplate2.get_swash_type() == SWASHPLATE_TYPE_H4_45) {
-            add_motor_num(CH_8);
-        }
-
-        // set rotor servo range
-        _main_rotor.init_servo();
-
     }
 
-    // reset swash servo range and endpoints
+    // init and setup all servos used in the swashplates
+    uint32_t mask = get_swashplate_servo_mask();
     for (uint8_t i=0; i<AP_MOTORS_HELI_DUAL_NUM_SWASHPLATE_SERVOS; i++) {
-        reset_swash_servo(SRV_Channels::get_motor_function(i));
-    }
-    if (_swashplate1.get_swash_type() == SWASHPLATE_TYPE_H4_90 || _swashplate1.get_swash_type() == SWASHPLATE_TYPE_H4_45) {
-        reset_swash_servo(SRV_Channels::get_motor_function(6));
-    }
-    if (_swashplate2.get_swash_type() == SWASHPLATE_TYPE_H4_90 || _swashplate2.get_swash_type() == SWASHPLATE_TYPE_H4_45) {
-        reset_swash_servo(SRV_Channels::get_motor_function(7));
+        if (mask & (1 << i)) {
+            if (!initialised_ok()) {
+                // map servo channels
+                add_motor_num(CH_1+i);
+            }
+
+            // reset swash servo range and endpoints
+            reset_swash_servo(SRV_Channels::get_motor_function(CH_1+i));
+        }
     }
 
-    set_initialised_ok(_frame_class == MOTOR_FRAME_HELI_DUAL);
```

Comment:
I have removed this, as do not think we ever setting initialised to true.  It is being set True, but then this: 
set_initialised_ok(frame_class == MOTOR_FRAME_HELI); 
in AP_MOTORSHELI (which will always be false because of the frame class) sets it false again